drivers/bridge/bridge.go

```diff
@@ -876,6 +877,55 @@ func setHairpinMode(nlh *netlink.Handle, link netlink.Link, enable bool) error {
 	return nil
 }
 
+func waitForIfaces(ifaceNames []string, timeout time.Duration, trigger func() error) error {
+	// Map interface names to their readiness states (false by default)
+	ifaceStates := make(map[string]bool)
+	allRunning := func() bool {
+		for _, ifaceState := range ifaceStates {
+			if !ifaceState {
+				return false
+			}
+		}
+		return true
+	}
+	for _, ifaceName := range ifaceNames {
+		ifaceStates[ifaceName] = false
+	}
+
+	// Start watching for link changes
+	notifyLink := make(chan netlink.LinkUpdate)
+	notifyDone := make(chan struct{})
+	netlink.LinkSubscribe(notifyLink, notifyDone)
+	defer close(notifyDone)
+
+	// Run the given function to trigger link changes
+	if err := trigger(); err != nil {
+		return err
+	}
+
+	// Do not return until the given interfaces are running, or timeout
+	timer := time.After(timeout)
+	for {
+		select {
+		case update := <-notifyLink:
+			if update.Header.Type == syscall.RTM_NEWLINK &&
+				update.IfInfomsg.Flags&syscall.IFF_RUNNING != 0 {
```

Comment:
I am not sure if this check is ever going to be true.

I need to double check, but I thought the interface flags would contain `syscall.IFF_RUNNING` only when both veth interfaces links are brought up, which is not true at this point in the code.

This makes me think the patch helps the unregister netdevice issue just by adding some delay in the interface programming flow, maybe making the kernel race less likely ?